ui/src/shared/components/time_series/TimeSeries.tsx

```diff
@@ -137,9 +144,10 @@ class TimeSeries extends PureComponent<Props, State> {
     if (
       this.props.uuid !== prevProps.uuid ||
       queriesDifferent ||
-      this.state.isFirstFetch
+      this.state.isFirstFetch ||
+      this.props.xPixels !== prevProps.xPixels
     ) {
-      this.executeQueries()
+      this.debouncer.call(this.executeQueries, EXECUTE_QUERIES_DEBOUNCE_MS)
```

Comment:
I feel uneasy putting this debounce logic in the `TimeSeries` component, which only exists since the `xPixels` prop can be updated dozens of times a second during a dashboard cell drag event. If it wasn't here, we would also be executing dozens of queries per second. 

I think a better solution would be to modify the parent, which provides `xPixels`, to only provide the prop as often as needed (basically to hoist the debounce logic). I found this a hard change to make though—it felt like I was hacking around the always-up-to-date and declarative rendering model in React. 

I would love if anyone has a better idea here.

Reply:
i agree it feels hacky to try to fight react's props / rendering model. however, here's something for consideration that i just found in a brief search: https://github.com/podefr/react-debounce-render.

i think what you've done here is fine. the HOC approach is also interesting.

another possibility is that you could disable querying during resizing, and possibly only trigger it upon completion of resizing.

i don't have further ideas but others may. cc @bthesorceror.